.github/workflows/testBuild.yml

```
@@ -259,24 +268,24 @@ jobs:
       # This step removes previous comments with links connected to the PR
       - name: maintain-comment
         uses: actions-cool/maintain-one-comment@de04bd2a3750d86b324829a3ff34d47e48e16f4b
+        if: ${{ fromJSON(needs.validateActor.outputs.READY_TO_BUILD) }}
         with:
           token: ${{ secrets.OS_BOTIFY_TOKEN }}
           body-include: 'Use the links below to test this build in android and iOS. Happy testing!'
           number: ${{ github.event.number || github.event.inputs.PULL_REQUEST_NUMBER }}
           delete: true
 
       - name: Publish links to apps for download
-        run: |
-          set -e
-          gh pr checkout "$PULL_REQUEST_NUMBER"
-          gh pr comment --body \
-          ":test_tube::test_tube: Use the links below to test this build in android and iOS. Happy testing! :test_tube::test_tube:
-          | android :robot:  | iOS :apple: |
-          | ------------- | ------------- |
-          | ${{fromJson(steps.set_var.outputs.android_paths).html_path}}  | ${{fromJson(steps.set_var.outputs.ios_paths).html_path}}  | 
-          | ![Android](https://api.qrserver.com/v1/create-qr-code/?size=200x200&data=${{fromJson(steps.set_var.outputs.android_paths).html_path}}) | ![iOS](https://api.qrserver.com/v1/create-qr-code/?size=200x200&data=${{fromJson(steps.set_var.outputs.ios_paths).html_path}}) |
-          | desktop :computer: | web :spider_web: |
-          | https://ad-hoc-expensify-cash.s3.amazonaws.com/desktop/$PULL_REQUEST_NUMBER/NewExpensify.dmg | https://$PULL_REQUEST_NUMBER.pr-testing.expensify.com |
-          | ![desktop](https://api.qrserver.com/v1/create-qr-code/?size=200x200&data=https://ad-hoc-expensify-cash.s3.amazonaws.com/desktop/$PULL_REQUEST_NUMBER/NewExpensify.dmg) | ![web](https://api.qrserver.com/v1/create-qr-code/?size=200x200&data=https://$PULL_REQUEST_NUMBER.pr-testing.expensify.com) |"
-        env:
+        if: ${{ fromJSON(needs.validateActor.outputs.READY_TO_BUILD) }}
+        uses: Expensify/App/.github/actions/javascript/postTestBuildComment@main
+        with:
+          PR_NUMBER: $PULL_REQUEST_NUMBER
```

NAB but I think it's a bit clearer to explicitly reference the `env` context so it's more clear where this variable comes from:

```suggestion
          PR_NUMBER: ${{ env.PULL_REQUEST_NUMBER }}
```